crates/rmf_site_picking/src/select/resources.rs

```diff
@@ -10,6 +11,10 @@ use web_time::Instant;
 #[derive(Default, Debug, Clone, Copy, Deref, DerefMut, Resource)]
 pub struct Selection(pub Option<Entity>);
 
+/// Used as a resource to keep track of selecting multiple entities
+#[derive(Default, Debug, Clone, Deref, DerefMut, Resource)]
+pub struct MultiSelection(pub HashSet<Entity>);
```

Comment:
I don't think we should fragment the selection pipeline like this. Let's just update the existing `Selection` resource to contain a `HashSet<Entity>` instead of an `Option<Entity>`, and we can leave it to downstream systems to decide what to do with the hash set.

In the interest of generalizing `Selection` I think we should change it into a struct, e.g.:

```rust
pub struct Selection {
    pub selected: HashSet<Entity>,
}
```

We should get rid of the `Deref` and `DerefMut` and instead provide helper functions:

```rust
impl Selection {
    pub fn get_single(&self) -> Option<Entity> {
        if self.selected.len() > 1 {
            return None;
        }

        self.selected.iter().next().copied()
    }
}
```

Reply:
Agreed. Having `get_single` as a method is a good idea and tidies up the use of `Selection` in other areas